src/content/docs/features/dialog.mdx

```diff
@@ -5,10 +5,315 @@ description: Native system dialogs for opening and saving files along with messa
+            // Initialize the plugin
+            .plugin(tauri_plugin_dialog::init())
+            .run(tauri::generate_context!())
+            .expect("error while running tauri application");
+    }
+    ```
+
+    3. If you'd like create dialogs in JavaScript, install the npm package as well:
+
+    <CommandTabs
+        npm="npm install @tauri-apps/plugin-dialog"
+        yarn="yarn add @tauri-apps/plugin-dialog"
+        pnpm="pnpm add @tauri-apps/plugin-dialog"
+    />
+
+</TabItem>
+</Tabs>
+
+## Usage
+
+Here's how you can use the dialog plugin:
+
+- [Create Yes/No Dialog](#create-yesno-dialog)
+- [Create Ok/Cancel Dialog](#create-okcancel-dialog)
+- [Create Message Dialog](#create-message-dialog)
+- [Open a File Selector Dialog](#open-a-file-selector-dialog)
+- [Save to File Dialog](#save-to-file-dialog)
+
+{/* TODO: Link to which language to use, frontend vs. backend guide when it's made */}
+
+The dialog plugin is available in both JavaScript and Rust.
+
+{/* ASK */}
+
+### Create Yes/No Dialog
+
+Shows a question dialog with `Yes` and `No` buttons
+
+<Tabs>
+<TabItem label="JavaScript">
+
+```js
+import { ask } from '@tauri-apps/plugin-dialog';
+
+// Create a Yes/No dialog
+const yes = await ask('This action cannot be reverted. Are you sure?', {
+	title: 'Tauri',
+	type: 'warning',
+});
+
+console.log(yes);
+// Prints boolean to the console
+```
+
+</TabItem>
+<TabItem label="Rust">
+
+<Stub />
+{/* TODO RUST: */}
+
+</TabItem>
+</Tabs>
+
+{/* CONFIRM */}
+
+### Create Ok/Cancel Dialog
+
+Shows a question dialog with `Ok` and `Cancel` buttons.
+
+<Tabs>
+<TabItem label="JavaScript">
+
+```js
+import { confirm } from '@tauri-apps/plugin-dialog';
+
+// Creates a confirmation Ok/Cancel dialog
+const confirmed = await confirm(
+	'This action cannot be reverted. Are you sure?',
+	{ title: 'Tauri', type: 'warning' }
+);
+
+console.log(confirmed);
+// Prints boolean to the console
+```
+
+</TabItem>
+<TabItem label="Rust">
+
+<Stub />
+{/* TODO RUST: */}
+
+</TabItem>
+</Tabs>
+
+{/* MESSAGE */}
+
+### Create Message Dialog
+
+Shows a message dialog with an `Ok` button.
+
+<Tabs>
+<TabItem label="JavaScript">
+
+```js
+import { message } from '@tauri-apps/plugin-dialog';
+
+// Shows message
+await message('File not found', { title: 'Tauri', type: 'error' });
+```
+
+</TabItem>
+<TabItem label="Rust">
+
+<Stub />
+{/* TODO RUST: */}
+
+</TabItem>
+</Tabs>
+
+{/* OPEN */}
+
+### Open a File Selector Dialog
+
+Open a file/directory selection dialog.
+
+<Tabs>    
+<TabItem label="JavaScript">
+
+```js
+import { open } from '@tauri-apps/plugin-dialog';
+
+// Open a dialog
+const file = await open();
+console.log(file);
+// Prints file data to the console
+```
+
+</TabItem>
+<TabItem label="Rust">
+
+<Stub />
+{/* TODO RUST: */}
+
+</TabItem>
+</Tabs>
+
+{/* OPEN > FILE */}
+
+#### File
+
+<Tabs>
+
+Select single file:
+
+<TabItem label="JavaScript">
+
+```js
+// Open a dialog
+const file = await open({
+	multiple: false,
```

Comment:
Instead of listing out each permutation of `multiple` and `directory` I think just adding a paragraph before the code block explaining what `multiple` and `directory` mean would be just as effective and reduce a lot of duplication.

Reply:
Ok, I reduced the duplication. Now, since there are multiple options, should those be explained as well? Not only the Open dialog, but Save, Message, and Confirm function, each have multiple options. 

Current Open dialog (The link to js api is inside the JavaScript tab)
<img src='https://github.com/tauri-apps/tauri-docs/assets/61759797/dadaf42c-b2d6-4473-9c30-fbeb9235ddc6' width='480'>

To compare with Save dialog
<img src='https://github.com/tauri-apps/tauri-docs/assets/61759797/db863f7f-2631-4bc6-8751-f8d1d5c0fefa' width='480'>

Reply:
I think how you've done in the "Open" one is perfect by linking to the API references. This lets the reader understand the very basic functionality (which is what this page is for) and let's them jump off to customise and configure it from there (which is what the reference page is for).

Reply:
I separated Rust and JavaScript guides into 2 blocks and moved stuff around. I hope that is ok. My reasoning is that this plugin have different ways to be used that does not translate into 1:1 blocks.

Also moved the link of Rust docs and JavaScript docs to the start of the block. Since it should be basic functionality I guess, content-wise, ready.